### PR TITLE
Unified the URL structure and supplemented validation

### PR DESCRIPTION
**Pull request by winniepukki**
Issue origins:
1. url-rewrites request gets sent before placeholders are loaded on the product breadcrumbs transition
2. the url structure is bad and unpredictable. we want to ensure that the behavior is OK for both cases if the URL is a string or an object and the structure is safe enough from falling out. it is also important to carry the state over

bugfix preview:
https://drive.google.com/file/d/15hlMRKFs2OdUISbnf8712aVJIJUcSEZj/view?usp=sharing

### CODE REVIEW

**Comment by winniepukki** [comment]
Scandiweb network speed data:
image

**Comment by aleksandrakorolova** [comment]
Home is written in caps lock in breadcrumbs 
image

Expected result:
image

**Comment by winniepukki** [comment]
The reason to use the `.toLoweCase()` method is that this issue appears to be a magic bug:
1. Changing the scss property, or making it `!important` does not help at all
2. Case becomes lower-cased on turning off and then turning back on the `text-transform` property in Chrome developer tools
3. Translated value is lower-cased

File: packages/scandipwa/src/component/Breadcrumbs/Breadcrumbs.component.js [hunk 1]
```diff
@@ -48,7 +48,8 @@ export class Breadcrumbs extends PureComponent {
     renderBreadcrumbList(breadcrumbs) {
         const breadcrumbsWithHome = [
             ...breadcrumbs,
-            { url: '/', name: __('Home') }
+            // Looks like a browser bug, temporary fixed with .toLowerCase()
+            { url: '/', name: __('Home').toLowerCase() }
```
Comment (yeegor) [inline packages/scandipwa/src/component/Breadcrumbs/Breadcrumbs.component.js:52]: Please figure this out. Very weird! Let's omit any crotches in our applications.

**Review by yeegor — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/Breadcrumbs/Breadcrumbs.component.js (hunk 1 above)
Comment (winniepukki) [reply packages/scandipwa/src/component/Breadcrumbs/Breadcrumbs.component.js]: Temporarily fixed as agreed

Comment (yeegor) [reply packages/scandipwa/src/component/Breadcrumbs/Breadcrumbs.component.js]: FYI this is probably related to the i18n rendering each letter as a separate node.

**Review by yeegor — APPROVED** [review]

**Pull request merged**